csharp/src/AddressBook/Addressbook.pb.cs

```diff
@@ -14,14 +14,36 @@ namespace Google.Protobuf.Examples.AddressBook {
   /// <summary>Holder for reflection information generated from addressbook.proto</summary>
   public static partial class AddressbookReflection {
 
+    internal const string ConstProtocVersion = "3.21.9";
+    /// <summary>Version of protoc used to generate this code.</summary>
+    public static global::System.Version ProtocVersion {
+      get { return protocVersion; }
+    }
+    private static readonly global::System.Version protocVersion = new global::System.Version(ConstProtocVersion);
+
+    /// <summary>Minimum version of Google.Protobuf which is compatible with this generated code.</summary>
+    public static global::System.Version MinimumRuntimeVersion {
+      get { return minimumRuntimeVersion; }
+    }
+    private static readonly global::System.Version minimumRuntimeVersion = new global::System.Version(3, 21, 9);
```

Comment:
A lot of duplicate Version instances are allocated and stay in memory as static fields. Can this be improved?

Reply:
It's two instances per original .proto file - which is a trivial addition compared with the amount used for descriptors etc. We could potentially have a cache somewhere, at the cost of complexity. Personally I'm not worried about this part - if we're looking for memory efficiency savings, there are other places I'd start.